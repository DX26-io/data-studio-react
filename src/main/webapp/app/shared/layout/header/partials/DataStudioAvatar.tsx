import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Button, Divider, Flex, Text, View } from '@adobe/react-spectrum';
import { Avatar } from '@material-ui/core';
import HeaderPopover from 'app/shared/layout/header/partials/HeaderPopover';
import { Translate } from 'react-jhipster';
import { useHistory } from 'react-router-dom';

const DataStudioAvatar = () => {

  const avatarStyles = makeStyles((theme) => ({
    small: {
      width: theme.spacing(4),
      height: theme.spacing(4)
    }
  }));
  const avatarClasses = avatarStyles();
  const history = useHistory();

  return (
    <>
      <HeaderPopover icon={<Avatar className={avatarClasses.small}><span className='avatar-initial'>H</span></Avatar>}>
        <View padding='size-300'>
          <Flex alignItems='center' justifyContent='center' direction='column'>
            <span className='spectrum-Body spectrum-Body--L'>
              <Text marginBottom='size-200'>
                <Translate contentKey="header.avatar.greeting">Hello</Translate> {'userName'}
              </Text>
            </span>
            <Divider marginY='size-200' size="S" />
            <Button isQuiet={true} variant='primary' marginBottom='size-200'>
              <Text>
                <Translate contentKey="header.avatar.preferences">Preferences</Translate>
              </Text>
            </Button>
            <Button onPress={() => history.push('/logout')} variant="primary" justifySelf='center'>
              <Text>
                <Translate contentKey="header.avatar.signOut">Sign Out</Translate>
              </Text>
            </Button>
          </Flex>
        </View>
      </HeaderPopover>
    </>
  );
};

export default DataStudioAvatar;
