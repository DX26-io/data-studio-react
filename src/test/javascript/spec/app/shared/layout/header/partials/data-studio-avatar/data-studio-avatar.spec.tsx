import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import DataStudioAvatar from 'app/shared/layout/header/partials/data-studio-avatar';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider as SpectrumProvider } from '@react-spectrum/provider';
import { defaultTheme } from '@adobe/react-spectrum';
import '@testing-library/jest-dom';

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: jest.fn(),
  }),
}));

describe('Data studio Avatar test', () => {
  let mountedWrapper;
  const initialState = {
    authentication: {
      account: {
        login: 'userName',
      },
    },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  const wrapper = () => {
    if (!mountedWrapper) {
      mountedWrapper = render(
        <SpectrumProvider theme={defaultTheme}>
          <Provider store={store}>
            <DataStudioAvatar />
          </Provider>
        </SpectrumProvider>
      );
    }
    return mountedWrapper;
  };

  const clickOnAvatarButton = tree => {
    const avatarButton = tree.getByTestId('avatarButton');
    userEvent.click(avatarButton);
  };

  it('should render component as expected', function () {
    const tree = wrapper();
    clickOnAvatarButton(tree);
    expect(tree.getByTestId('userGreeting')).toBeDefined();
    expect(tree.getByTestId('preferencesButton')).toBeDefined();
    expect(tree.getByTestId('logoutButton')).toBeDefined();
  });

  it('should render user name correctly', function () {
    const tree = wrapper();
    clickOnAvatarButton(tree);
    expect(tree.getByTestId('userGreeting')).toHaveTextContent('userName');
  });
});
