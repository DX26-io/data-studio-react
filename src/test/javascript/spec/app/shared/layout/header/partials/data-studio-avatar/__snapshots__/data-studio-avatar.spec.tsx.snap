// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data studio Avatar test should render component as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <DataStudioAvatar>
    <$e8eede229dc4bb152d31b6896bc207be$var$DialogTrigger
      type="popover"
    >
      <$e8eede229dc4bb152d31b6896bc207be$var$PopoverTrigger
        content={
          <ForwardRef($c24df1e30ea6d5a25486b1eedd756cd$var$Dialog)>
            <ForwardRef($b1630a1ad5492629b7ba4ec7eeabf6d2$var$Content)>
              <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
                alignItems="center"
                direction="column"
                gap="size-175"
                justifyContent="center"
              >
                <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                  <span
                    className="spectrum-Body spectrum-Body--sizeL"
                  >
                    <Translate
                      component="span"
                      contentKey="header.avatar.greeting"
                    >
                      Hello
                    </Translate>
                    <strong>
                       userName
                    </strong>
                  </span>
                </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                <ForwardRef($e4a3649e5820982ed2d66266f2$var$Divider)
                  size="M"
                />
                <ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)
                  isQuiet={true}
                  variant="primary"
                >
                  <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                    <Translate
                      component="span"
                      contentKey="header.avatar.preferences"
                    >
                      Preferences
                    </Translate>
                  </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                </ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)>
                <ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)
                  justifySelf="center"
                  onPress={[Function]}
                  variant="primary"
                >
                  <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                    <Translate
                      component="span"
                      contentKey="header.avatar.signOut"
                    >
                      Sign Out
                    </Translate>
                  </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                </ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)>
              </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
            </ForwardRef($b1630a1ad5492629b7ba4ec7eeabf6d2$var$Content)>
          </ForwardRef($c24df1e30ea6d5a25486b1eedd756cd$var$Dialog)>
        }
        state={
          Object {
            "close": [Function],
            "isOpen": false,
            "open": [Function],
            "toggle": [Function],
          }
        }
        trigger={
          <ForwardRef($f3063a10f4b2c05148061fca201b$var$ActionButton)
            aria-label="User avatar"
            isQuiet={true}
            marginEnd="size-200"
          >
            <User
              size="M"
            />
          </ForwardRef($f3063a10f4b2c05148061fca201b$var$ActionButton)>
        }
      >
        <$e8eede229dc4bb152d31b6896bc207be$var$DialogTriggerBase
          dialogProps={
            Object {
              "id": "react-aria-6430236964-1",
            }
          }
          overlay={
            <ForwardRef($cb5ac2f1b271cf568ded3aa34c88ecd$var$Popover)
              UNSAFE_style={
                Object {
                  "maxHeight": undefined,
                  "position": "absolute",
                  "zIndex": 100000,
                }
              }
              arrowProps={
                Object {
                  "style": Object {
                    "left": undefined,
                    "top": undefined,
                  },
                }
              }
              isOpen={false}
              onClose={[Function]}
            >
              <ForwardRef($c24df1e30ea6d5a25486b1eedd756cd$var$Dialog)>
                <ForwardRef($b1630a1ad5492629b7ba4ec7eeabf6d2$var$Content)>
                  <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
                    alignItems="center"
                    direction="column"
                    gap="size-175"
                    justifyContent="center"
                  >
                    <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                      <span
                        className="spectrum-Body spectrum-Body--sizeL"
                      >
                        <Translate
                          component="span"
                          contentKey="header.avatar.greeting"
                        >
                          Hello
                        </Translate>
                        <strong>
                           userName
                        </strong>
                      </span>
                    </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                    <ForwardRef($e4a3649e5820982ed2d66266f2$var$Divider)
                      size="M"
                    />
                    <ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)
                      isQuiet={true}
                      variant="primary"
                    >
                      <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                        <Translate
                          component="span"
                          contentKey="header.avatar.preferences"
                        >
                          Preferences
                        </Translate>
                      </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                    </ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)>
                    <ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)
                      justifySelf="center"
                      onPress={[Function]}
                      variant="primary"
                    >
                      <ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                        <Translate
                          component="span"
                          contentKey="header.avatar.signOut"
                        >
                          Sign Out
                        </Translate>
                      </ForwardRef($fe1955053990ab873941970c7463$var$Text)>
                    </ForwardRef($ead2ca032915257adf53f639d99d1a73$var$Button)>
                  </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
                </ForwardRef($b1630a1ad5492629b7ba4ec7eeabf6d2$var$Content)>
              </ForwardRef($c24df1e30ea6d5a25486b1eedd756cd$var$Dialog)>
            </ForwardRef($cb5ac2f1b271cf568ded3aa34c88ecd$var$Popover)>
          }
          state={
            Object {
              "close": [Function],
              "isOpen": false,
              "open": [Function],
              "toggle": [Function],
            }
          }
          trigger={
            <ForwardRef($f3063a10f4b2c05148061fca201b$var$ActionButton)
              aria-label="User avatar"
              isQuiet={true}
              marginEnd="size-200"
            >
              <User
                size="M"
              />
            </ForwardRef($f3063a10f4b2c05148061fca201b$var$ActionButton)>
          }
          triggerProps={
            Object {
              "aria-controls": null,
              "aria-expanded": false,
              "aria-haspopup": undefined,
              "ref": Object {
                "current": <button
                  aria-expanded="false"
                  aria-label="User avatar"
                  class="_spectrum-ActionButton_7a745 _spectrum-ActionButton--quiet_7a745"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="_spectrum-Icon_8ac98 _spectrum-Icon--sizeM_8ac98 _spectrum-Icon_7a745"
                    focusable="false"
                    role="img"
                    viewBox="0 0 36 36"
                  >
                    <path
                      d="M32.949 34a.993.993 0 0 0 1-1.053c-.661-7.184-8.027-9.631-10.278-9.827C22.026 22.977 22 21.652 22 20c0 0 3.532-3.943 3.532-8.958C25.532 5.617 22.445 2 18 2s-7.532 3.617-7.532 9.042C10.468 16.057 14 20 14 20c0 1.652-.026 2.977-1.674 3.12-2.251.2-9.617 2.643-10.278 9.827a.993.993 0 0 0 1 1.053z"
                    />
                  </svg>
                </button>,
              },
            }
          }
          type="popover"
        >
          <ForwardRef
            aria-controls={null}
            aria-expanded={false}
            isPressed={false}
            onPress={[Function]}
          >
            <ForwardRef($f3063a10f4b2c05148061fca201b$var$ActionButton)
              aria-label="User avatar"
              isQuiet={true}
              marginEnd="size-200"
            >
              <FocusRing
                focusRingClass="_focus-ring_7a745"
              >
                <button
                  aria-controls={null}
                  aria-expanded={false}
                  aria-label="User avatar"
                  className="_spectrum-ActionButton_7a745 _spectrum-ActionButton--quiet_7a745"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchCancel={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  style={
                    Object {
                      "marginRight": "var(--spectrum-global-dimension-size-200, var(--spectrum-alias-size-200))",
                    }
                  }
                  type="button"
                >
                  <SlotProvider
                    slots={
                      Object {
                        "icon": Object {
                          "UNSAFE_className": "_spectrum-Icon_7a745",
                          "size": "S",
                        },
                        "text": Object {
                          "UNSAFE_className": "_spectrum-ActionButton-label_7a745",
                        },
                      }
                    }
                  >
                    <User
                      size="M"
                    >
                      <Icon
                        size="M"
                      >
                        <A4uUser
                          aria-hidden={true}
                          className="_spectrum-Icon_8ac98 _spectrum-Icon--sizeM_8ac98 _spectrum-Icon_7a745"
                          focusable="false"
                          role="img"
                          style={Object {}}
                        >
                          <svg
                            aria-hidden={true}
                            className="_spectrum-Icon_8ac98 _spectrum-Icon--sizeM_8ac98 _spectrum-Icon_7a745"
                            focusable="false"
                            role="img"
                            style={Object {}}
                            viewBox="0 0 36 36"
                          >
                            <path
                              d="M32.949 34a.993.993 0 0 0 1-1.053c-.661-7.184-8.027-9.631-10.278-9.827C22.026 22.977 22 21.652 22 20c0 0 3.532-3.943 3.532-8.958C25.532 5.617 22.445 2 18 2s-7.532 3.617-7.532 9.042C10.468 16.057 14 20 14 20c0 1.652-.026 2.977-1.674 3.12-2.251.2-9.617 2.643-10.278 9.827a.993.993 0 0 0 1 1.053z"
                            />
                          </svg>
                        </A4uUser>
                      </Icon>
                    </User>
                  </SlotProvider>
                </button>
              </FocusRing>
            </ForwardRef($f3063a10f4b2c05148061fca201b$var$ActionButton)>
          </ForwardRef>
          <ForwardRef($cb5ac2f1b271cf568ded3aa34c88ecd$var$Popover)
            UNSAFE_style={
              Object {
                "maxHeight": undefined,
                "position": "absolute",
                "zIndex": 100000,
              }
            }
            arrowProps={
              Object {
                "style": Object {
                  "left": undefined,
                  "top": undefined,
                },
              }
            }
            isOpen={false}
            onClose={[Function]}
          >
            <ForwardRef($d0270a5a2198467ee027b9df554cc09$var$Overlay)
              UNSAFE_style={
                Object {
                  "maxHeight": undefined,
                  "position": "absolute",
                  "zIndex": 100000,
                }
              }
              isOpen={false}
            />
          </ForwardRef($cb5ac2f1b271cf568ded3aa34c88ecd$var$Popover)>
        </$e8eede229dc4bb152d31b6896bc207be$var$DialogTriggerBase>
      </$e8eede229dc4bb152d31b6896bc207be$var$PopoverTrigger>
    </$e8eede229dc4bb152d31b6896bc207be$var$DialogTrigger>
  </DataStudioAvatar>
</Provider>
`;
